Userland/Applications/Browser/History.cpp

```diff
@@ -52,20 +55,28 @@ void History::clear()
     m_current = -1;
 }
 
-const Vector<URL> History::get_back_history()
+void History::update_title(const URL& url, String title)
 {
-    Vector<URL> back_history;
+    for (auto& item : m_items) {
+        if (item.url == url)
+            item.title = title;
+    }
```

Comment:
Why does this need to update *all* matching urls in the history - last item only seems sufficient, no match needed? At least the way this is currently used it should only alter the title of the history entry for the current page, and I would't include previous visits with possibly different titles in that.

Reply:
I had some issues with the wrong title getting set when no checks were made, but testing it now it seems that
```
    if (m_items[m_current].url == url)
        m_items[m_current].title = title;
```
is enough for it to work correctly